contrib/cindercompatibleapi/converter/volume.go

```diff
@@ -0,0 +1,380 @@
+	resp.Volume.AvailabilityZone = volume.AvailabilityZone
+	resp.Volume.UpdatedAt = volume.BaseModel.UpdatedAt
+	resp.Volume.ID = volume.BaseModel.Id
+	resp.Volume.Size = volume.Size
+	resp.Volume.UserID = volume.UserId
+	resp.Volume.Metadata = make(map[string]string)
+	resp.Volume.Status = volume.Status
+	resp.Volume.Description = volume.Description
+	resp.Volume.Name = volume.Name
+	resp.Volume.CreatedAt = volume.BaseModel.CreatedAt
+
+	return &resp
+}
+
+// *******************List accessible volumes*******************
+
+// ListVolumesRespSpec ...
+type ListVolumesRespSpec struct {
+	Volumes []ListRespVolume `json:"volumes"`
+	Count   int64            `json:"count,omitempty"`
+}
+
+// ListRespVolume ...
+type ListRespVolume struct {
+	ID    string `json:"id"`
+	Links []Link `json:"links,omitempty"`
+	Name  string `json:"name"`
+}
+
+// ListVolumesResp ...
+func ListVolumesResp(volumes []*model.VolumeSpec) *ListVolumesRespSpec {
+	var resp ListVolumesRespSpec
+	var cinderVolume ListRespVolume
+
+	if 0 == len(volumes) {
+		resp.Volumes = make([]ListRespVolume, 0, 0)
+	} else {
+		for _, volume := range volumes {
+			cinderVolume.ID = volume.Id
+			cinderVolume.Name = volume.Name
+
+			resp.Volumes = append(resp.Volumes, cinderVolume)
+		}
+	}
+
+	return &resp
+}
+
+// *******************Show a volume’s details*******************
+
+// ShowVolumeRespSpec ...
+type ShowVolumeRespSpec struct {
+	Volume ShowRespVolume `json:"volume"`
+}
+
+// ShowRespVolume ...
+type ShowRespVolume struct {
+	MigrationStatus     string            `json:"migration_status,omitempty"`
+	Attachments         []RespAttachment  `json:"attachments"`
+	Links               []Link            `json:"links,omitempty"`
+	AvailabilityZone    string            `json:"availability_zone,omitempty"`
+	Host                string            `json:"os-vol-host-attr:host,omitempty"`
+	Encrypted           bool              `json:"encrypted ,omitempty"`
+	UpdatedAt           string            `json:"updated_at"`
+	ReplicationStatus   string            `json:"replication_status,omitempty"`
+	SnapshotID          string            `json:"snapshot_id,omitempty"`
+	ID                  string            `json:"id"`
+	Size                int64             `json:"size"`
+	UserID              string            `json:"user_id"`
+	TenantID            string            `json:"os-vol-tenant-attr:tenant_id,omitempty"`
+	Migstat             string            `json:"os-vol-mig-status-attr:migstat,omitempty"`
+	Metadata            map[string]string `json:"metadata"`
+	Status              string            `json:"status"`
+	VolumeImageMetadata map[string]string `json:"volume_image_metadata,omitempty"`
+	Description         string            `json:"description"`
+	Multiattach         bool              `json:"multiattach,omitempty"`
+	SourceVolID         string            `json:"source_volid,omitempty"`
+	ConsistencygroupID  string            `json:"consistencygroup_id,omitempty"`
```

Comment:
the same

Reply:
In order to comply with golint's rules, "Id" must be written in all capitals.